tensorflow/python/ops/array_ops.py

```diff
@@ -125,6 +125,32 @@ def shape(input, name=None):
     else:
       return gen_array_ops.shape(input, name=name)
 
+def size(input, name=None):
+  """Returns the size of a tensor.
+
+  This operation returns an integer representing the number of elements in
+  `input`.
+
+  For example:
+
+  ```python
+  # 't' is [[[1, 1,, 1], [2, 2, 2]], [[3, 3, 3], [4, 4, 4]]]]
+  size(t) ==> 12
+  ```
+
+  Args:
+    input: A `Tensor` or `SparseTensor`.
+    name: A name for the operation (optional).
+
+  Returns:
+    A `Tensor` of type `int32`.
+  """
+  with ops.op_scope([input], name, "Size") as name:
+    if isinstance(input, ops.SparseTensor):
+      return gen_math_ops._prod(input.shape, 0, name=name)
```

Comment:
This may be wrong, but I didn't do it because it may lead to cyclic dependencies. `math_ops.py` imports `array_ops`, so importing `math_ops` here will lead to a cycle. I'm not sure how dependencies work in Python, so this may be entirely wrong. Also, `tf.reduce_prod()` uses `gen_math_ops._prod()` within the implementation, so I guess it doesn't make a difference.